mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -156,6 +156,20 @@ class KMeans private (
     this
   }
 
+  // Initial cluster centers can be provided as a KMeansModel object rather than using the
+  // random or k-means|| initializationMode
+  private var initialModel: Option[KMeansModel] = None
+
+  /** Set the initial starting point, bypassing the random initialization or k-means||
+    * The condition (model.k == this.k) must be met; failure will result in an
+    * IllegalArgumentException.
+    */
+  def setInitialModel(model: KMeansModel): this.type = {
+    require(model.k==k, "mismatched cluster count")
```

Comment:
Missing spaces around `==` here. You might print the actual vs expected in the message.